system/lib/compiler-rt/lib/asan/asan_emscripten.cpp

```diff
@@ -18,6 +20,15 @@ void InitializeShadowMemory() {
   // Poison the shadow memory of the shadow area at the start of the address
   // space. This helps catching null pointer dereference.
   FastPoisonShadow(kLowShadowBeg, kLowShadowEnd - kLowShadowBeg, 0xff);
+
+  // Assert that the shadow region is large enough.  We don't want to start
+  // running into the static data region which starts right after the shadow
+  // region.
+  uptr max_address = (__builtin_wasm_memory_size(0) * WASM_PAGE_SIZE) - 1;
```

Comment:
This will overflow in wasm32 if memory size is 4GB, I think.

Possible fix:

```suggestion
  uptr max_address = (__builtin_wasm_memory_size(0) * uint64_t(WASM_PAGE_SIZE)) - 1;
```

Reply:
Done